packages/react-scripts/config/webpack.config.prod.js

```diff
@@ -222,22 +242,38 @@ module.exports = {
                 },
                 {
                   loader: require.resolve('postcss-loader'),
+                  options: postCSSLoaderOptions,
+                },
+              ],
+            },
+            extractTextPluginOptions
+          )
+        ),
+        // Note: this won't work without `new ExtractTextPlugin()` in `plugins`.
+      },
+      // Adds support for CSS Modules (https://github.com/css-modules/css-modules)
+      // using the extension .module.css
+      {
+        test: /\.module\.css$/,
+        loader: ExtractTextPlugin.extract(
+          Object.assign(
+            {
+              fallback: require.resolve('style-loader'),
+              use: [
+                {
+                  loader: require.resolve('css-loader'),
                   options: {
-                    ident: 'postcss', // https://webpack.js.org/guides/migrating/#complex-options
-                    plugins: () => [
-                      require('postcss-flexbugs-fixes'),
-                      autoprefixer({
-                        browsers: [
-                          '>1%',
-                          'last 4 versions',
-                          'Firefox ESR',
-                          'not ie < 9', // React doesn't support IE8 anyway
-                        ],
-                        flexbox: 'no-2009',
-                      }),
-                    ],
+                    importLoaders: 1,
+                    minimize: true,
+                    sourceMap: true,
+                    modules: true,
+                    localIdentName: '[name]__[local]___[hash:base64:5]',
```

Comment:
True that gzip will lessen/eliminate the impact of really long classnames on file transfer size, I'd still prefer to have smaller, more readable names. `[name]__[local]` is very useable in e.g. devtools, and gzip will not help that situation.

My personal preference would be to use `getLocalIdent` to construct something like `[name]__[local]__[hash of path]` so that it's readable, consistent between builds even if content changes, and will be reasonably short (as long as your component names and class names aren't crazy long).

IMO a hash based on the path to the file makes more sense than the hash based on the contents of the CSS class. The hash of the contents only makes since if we want to 
 cachebust at the classname level (we don't), or if we expect that multiple unrelated classes  that have the same content could hash to the same name (only makes sense if our localIdentName is _only_ the hash, but given that we're using `[name]__[local]` we're basically preventing this).